net/ddns-scripts/files/usr/lib/ddns/update_one_com.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/sh
+
+# ONE.COM DDNS SCRIPT
+# REQUIRES CURL
+# $ opkg install curl
+
+# SCRIPT BY LUGICO
+# CONTACT: main@lugico.de
+
+[ -z "$CURL" ] && [ -z "$CURL_SSL" ] && write_log 14 "one.com communication require cURL with SSL support. Please install"
+[ -z "$domain" ]   && write_log 14 "Service section not configured correctly! Missing 'domain'"
+[ -z "$username" ] && write_log 14 "Service section not configured correctly! Missing 'username'"
+[ -z "$password" ] && write_log 14 "Service section not configured correctly! Missing 'password'"
+
+write_log 0 "one.com ddns script started"
+
+local __SUBDOMAIN __MAINDOMAIN __LOGINURL __RECORDID
+local __TTL=3600
+
+COOKIEJAR=$(mktemp /tmp/one_com_cookiejar.XXXXXX) || exit 1
+
+__SUBDOMAIN=$(echo $domain | sed -e 's/[^\.]*\.[^\.]*$//' -e 's/\.$//' )
+__MAINDOMAIN=$(echo $domain | sed -e "s/${__SUBDOMAIN}\.//" )
+
+
+# LOGGING IN
+# GET LOGIN POST URL FROM FORM
+__LOGINURL=$( $CURL \
+	-RsSL \
+	--stderr $ERRFILE \
+	-c $COOKIEJAR \
+	"https://www.one.com/admin/" \
+	| grep 'Login-form login autofill' \
+	| sed -e 's/.*action="//' -e 's/".*//' -e 's/\&amp;/\&/g' \
+)
+
+# POST LOGIN DATA
+$CURL \
+	-RsSL \
+	--stderr $ERRFILE \
+	-c $COOKIEJAR \
+	-b $COOKIEJAR \
+	"${__LOGINURL}" \
+	-H "Content-Type: application/x-www-form-urlencoded" \
+	-X POST \
+	-d "username=${username}&password=${password}&credentialId=" \
+	| grep "Invalid username or password." > $DATFILE
+
+if [ "$?" == "0" ] ; then
+	write_log 14 "Invalid credentials"
+	return 1
+fi
+
+
+# SETTING DOMAIN
+$CURL -RsSL \
+	--stderr $ERRFILE \
+	-c $COOKIEJAR \
+	-b $COOKIEJAR \
+	"https://www.one.com/admin/select-admin-domain.do?domain=${__MAINDOMAIN}" \
+	| grep "<meta name=\"one.com:active-domain\" content=\"${__MAINDOMAIN}\"/>" > $DATFILE
+
+if [ "$?" != "0" ] ; then
+	write_log 14 "Failed to select domain '${__MAINDOMAIN}'"
+	return 1
+fi
+
+
+# GETTING RECORD ID
+__RECORDID=$( $CURL \
+	-RsSL \
+	--stderr $ERRFILE \
+	-c $COOKIEJAR \
+	-b $COOKIEJAR \
+	"https://www.one.com/admin/api/domains/${__MAINDOMAIN}/dns/custom_records" \
+	| sed 's/,/\n/g' \
+	| while read line ; do
```

Comment:
I don't think that's very legible either! can't we rewrite that.

I understand your approach, that you process the data immediately without storing it in between.
How many lines are we talking about?
The maintenance of the code is not good. I was already willing to blame the last line with the `done \`.

Reply:
The cURL request returns a single JSON line with all the custom DNS records the user has created. It can range anywhere from just a few to multiple thousands, though that's rather unlikely I think. It should be safe to store the data for a bit instead of immediately processing it. This was just my first solution.

I didn't notice the backslash in `done \` was unnecessary, I'll remove it.

Reply:
If this is a json, then we can also process this with json_init?
Search for this string in the code and hopefully you will understand what I mean.

Reply:
Thanks for the idea. The jshn library is a lot more elegant, I wasn't aware that even existed. Let me know if I can improve anything on my implementation of it.